File: algorithms/sorting/bubble-sort/BubbleSort.dart
class Sorting {
  static List<int> BubbleSort(List<int> unsortedList) {
    bool swapped;
    int len = unsortedList.length;
    int temp;

    for (int i = 0; i < len; i++) {
      swapped = false;

      for (int j = 0; j < len; j++) {
        if (unsortedList[j] > unsortedList[j + 1]) {
          temp = unsortedList[j];
          unsortedList[j] = unsortedList[j + 1];
          unsortedList[j + 1] = temp;
          swapped = true;
        }
      }

      if (!swapped) {
        break;
      }
    }

    return unsortedList;
  }
}

File: algorithms/sorting/bubble-sort/BubbleSort.ts
class Sorting {
	public BubbleSort(unsortedArray: number[]): number[] {
		let swapped: boolean;
		const length: number = unsortedArray.length;

		for (let i: number = 0; i < length; i++) {
			swapped = false;

			for (let j: number = 0; j < length; j++) {
				if (unsortedArray[j] > unsortedArray[j + 1]) {
					const temp: number = unsortedArray[j];
					unsortedArray[j] = unsortedArray[j + 1];
					unsortedArray[j + 1] = temp;
					swapped = true;
				}

				if (!swapped) break;
			}
		}

		return unsortedArray;
	}
}

export { Sorting };
